nifi-extension-bundles/nifi-provenance-repository-bundle/nifi-persistent-provenance-repository/src/main/java/org/apache/nifi/provenance/util/StorageSummaryEvent.java

```diff
@@ -19,11 +19,14 @@
 
 import java.util.List;
 import java.util.Map;
+import java.util.Set;
+
 import org.apache.nifi.provenance.ProvenanceEventRecord;
 import org.apache.nifi.provenance.ProvenanceEventType;
+import org.apache.nifi.provenance.UpdateableProvenanceEventRecord;
 import org.apache.nifi.provenance.serialization.StorageSummary;
 
-public class StorageSummaryEvent implements ProvenanceEventRecord {
+public class StorageSummaryEvent implements UpdateableProvenanceEventRecord {
```

Comment:
This change to an 'UpdateableProvenanceEventRecord' abstraction and addition of setters stands out.  What is it about this new concept (previous event identifiers) that is causing a change in abstraction ?